es5-shim.js

```diff
@@ -329,6 +360,97 @@ var strSplit = call.bind(StringPrototype.split);
 var strIndexOf = call.bind(StringPrototype.indexOf);
 var push = call.bind(array_push);
 
+// ES-5 15.3.4.4
+// http://es5.github.io/#x15.3.4.4
+var hasToStringTagBasicBug = !(toStr() === '[object Undefined]' && toStr(null) === '[object Null]');
+var hasToStringTagArgumentsBug = toStr(function () {
+  return arguments;
+}) !== '[object Arguments]';
```

Comment:
Yep, but it's different in different environments. All ES5 environments give `[object Arguments]`, whether in the spec or not though I'm not sure. Can't use your check as it will cause recursion in the fix.